http4k-core/src/main/kotlin/org/http4k/filter/ResponseFilters.kt

```diff
@@ -131,6 +135,74 @@ object ResponseFilters {
     fun Base64EncodeBody() = Filter { next ->
         { next(it).run { body(Base64.getEncoder().encodeToString(body.payload.array())) } }
     }
+
+    class EtagSupport : Filter {
+        private val md5 = ThreadLocal.withInitial {
+            try {
+                MessageDigest.getInstance("MD5")
+            } catch (e: NoSuchAlgorithmException) {
```

Comment:
Why would this need catching? Would it make any difference as this is not Java anymore?

Reply:
It's a leftover from before the optimisation to make it a supplier to a threadlocal went in. Feel free to drop it.